### PR TITLE
[CIS-795] Remove defaultAppearance Hook

### PR DESCRIPTION
**Pull request by DominikBucher12**
# Motivation for PR:
- `DefaultAppearance` hook usage was ambigous and we agreed that we should delete the implementation because we cannot **restrict the usage of this method only to modify appearance** and not actually code for layout and other properties which could potentially lead to side effects because of the nature of `StreamChat._View` lifecycle. 

# What this PR does: 
- Removes `Customizable.defaultAppearance` method which was responsible for default `StreamChat` appearance. With this it also removes `defaultAppearance hook`  which was the root motivation for this PR. Instead of using `defaultAppearance` in our codebase, we switch to overriding `setUpAppearance` with `super.setUpAppearance` calls.
# How to test it
- The biggest prove should be the tests being green. Nevertheless always try to run the app and take a look if something is behaving strange or is different layout then before, if yes we should fix it and add test to it as it's not covered. 

Please check the code snippets and documentation if I should improve something or is not clear.

### CODE REVIEW

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift [hunk 1]
```diff
@@ -104,7 +104,7 @@ open class _ChatChannelListVC<ExtraData: ExtraDataTypes>: _ViewController,
         loadingIndicator.pin(anchors: [.centerX, .centerY], to: view)
     }
     
-    override public func defaultAppearance() {
+    override open func setUpAppearance() {
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:107]: We need to call `super` for `setUpAppearance` now

Comment (DominikBucher12) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:107]: ✅

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView_Tests.swift [hunk 2]
```diff
@@ -50,21 +50,6 @@ class ChatChannelListItemView_Tests: XCTestCase {
...
-        }
-        
-        let view = TestView().withoutAutoresizingMaskConstraints
-        
-        view.addSizeConstraints()
-        
-        view.content = content
-        AssertSnapshot(view)
-    }
-
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView_Tests.swift:67]: We need to remove all snapshot files that have `_usingAppearanceHook` and are now obsolete 😄

Comment (DominikBucher12) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView_Tests.swift:67]: Yes :D Deleted :)

**Review by VojtaStavik — APPROVED** [review]
LGTM once you fix the snapshot zero-size snapshot tests 👍

File: Sources/StreamChatUI/ChatMessageList/ChatChannelVC.swift [hunk 3]
```diff
@@ -34,9 +34,9 @@ open class _ChatChannelVC<ExtraData: ExtraDataTypes>: _ChatVC<ExtraData> {
         return navbarListener
     }
 
-    override public func defaultAppearance() {
-        super.defaultAppearance()
-
+    override open func setUpAppearance() {
+        super.setUpAppearance()
+        super.setUpAppearance()
```
Comment (skorepak) [inline Sources/StreamChatUI/ChatMessageList/ChatChannelVC.swift:39]: Double super 👀

**Review by skorepak — APPROVED** [review]

File: Sources/StreamChatUI/ChatMessageList/ChatChannelVC.swift (hunk 3 above)
Comment (DominikBucher12) [reply Sources/StreamChatUI/ChatMessageList/ChatChannelVC.swift]: Fixed.

File: Sources/StreamChatUI/CommonViews/BaseViews.swift [hunk 4]
```diff
@@ -76,12 +76,11 @@ open class _View: UIView, AppearanceSetting, Customizable {
         
         setUp()
         setUpLayout()
-        (self as! Self).applyDefaultAppearance()
+
```
Comment (DominikBucher12) [inline Sources/StreamChatUI/CommonViews/BaseViews.swift]: Deleted as well 😅

**Pull request merged**